aerogear/mobile-security-service-operator
# fix(groupfiles - codegen) : Fix required regards the change in the Issue 1812

## Motivation
- https://github.com/operator-framework/operator-sdk/issues/1812
- https://github.com/operator-framework/operator-sdk/pull/1822

## What
- Fix group name in the files, crd's and imports


## Why
- Fix `make code/gen` command

## How
- Rename `/mobile-security-service-operator/pkg/apis/` to use the correct name of the group : `mobile-security-service`
- Add group.go in API dir based on the `group.go` template
- Fix  import paths in the files to looking for in the group path with a hyphen
- Remove duplicated CRD files generated previously
- Remane `cluster_role_binding.yaml` to `role_binding.yaml` and `cluster_role.yaml` to `role.yaml` in order to keep the names compatibles with what is expected in the operator-framework and allow the tool to work with. 
 
## Verification Steps
 
1. Check if the CI finishes with success
2. Check the project by running it locally with the image `cmacedo/mobile-security-service-operator:fix-code-gen-issue-sdk` 

## Checklist:


## Progress


- Wait for the https://github.com/operator-framework/operator-sdk/pull/1822 be merged and released then update the .circleci/config.yaml to use the operator sdk version released with. (**IMPORTANT**: maybe it will be solved in another way then this PR should be closed/ignored)

## Additional Notes
